devito/compiler.py

```diff
@@ -354,7 +354,8 @@ def __init__(self, *args, **kwargs):
         super(GNUCompiler, self).__init__(*args, **kwargs)
 
         self.cflags += ['-march=native', '-Wno-unused-result', '-Wno-unused-variable',
-                        '-Wno-unused-but-set-variable', '-ffast-math']
+                        '-Wno-unused-but-set-variable', '-ffast-math',
+                        '-fno-unsafe-math-optimizations']
```

Comment:
hang on -- what's the performance penalty with this?

Reply:
maybe we should add DEVITO_SAFEMATH escape hatch rather than adding it unconditionally

Reply:
how about the other compilers?

Reply:
There are lots of health warnings relating to the use of -funsafe-math-optimization ... because it's unsafe. I think the default should be -fno-unsafe-math-optimization and we should add a Devito compiler optimization option that specifically enables unsafe optimization options.

Reply:
DEVITO_SAFE_MATH configuration option added. The default is 0 (off).